Web/Server-Side Request Forgery.md

```diff
@@ -0,0 +1,60 @@
+
+## Capabilities and Risk
+
+- By exploiting SSRF, an attacker can make requests from the application server.
+- An attacker can interact with otherwise restricted IP addresses and services, either on the server itself (localhost), or on other IPs. This can give an external attacker visibility to an internal environment.  This includes using the vulnerable server to port scan other hosts (Cross Site Port Attacks (XSPA)). 
+- If the vulnerable server can communicate with backend API's or services that do not require authentication, the external attacker can fully interact with those services.  
+- If the vulnerable application is hosted in a cloud environment, such as Amazon EC2 and OpenStack, this may allow the attacker to gain access to metadata services, which can be used to gain access to sensitive information, sometimes including credentials or private keys.  
+
+## Detection
+
+1) Browse the target application using an intercepting proxy (Burp, Fiddler, ZAP, etc). Determine if the following conditions apply:
+  - The target application is accepting a URL from you.  Ex: www.thirdpartysite.com
+  - The target application is displaying part or all of the result back to you.
+
+2) If both conditions apply, look at your proxy logs.  If you do not see the request to the resource (www.thirdpartysite.com) in your proxy logs, but you see the content on the page, this indicates that the content returned to you has been requested by the server itself on your behalf. This behavior indicates the application is vulnerable to SSRF.    
+
+## Remediation
+
+Rather than proxying requests on behalf of users, the application should have the user’s browser retrieve the desired information. If it is necessary to proxy the request, a whitelist should be used on the server side and the User-Agent information should be stripped or modified.        
+
+## References
+
+- [OWASP Top 10 2013-A1-Injection](https://www.owasp.org/index.php/Top_10_2013-A1-Injection)
+- [CWE-918: Server-Side Request Forgery (SSRF)](http://cwe.mitre.org/data/definitions/918.html)
+- [CWE-441: Unintended Proxy or Intermediary ('Confused Deputy')](http://cwe.mitre.org/data/definitions/441.html)
+- [EC2 Instance Metadata Service Documentation](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html)
+- [OpenStack Metadata Service Documentation](http://docs.openstack.org/admin-guide-cloud/content/section_metadata-service.html)
+- [Compromising an unreachable Solr server with CVE-2013-6397](http://www.agarri.fr/kom/archives/2013/11/27/compromising_an_unreachable_solr_server_with_cve-2013-6397/index.html)
+- [Bringing a Machete to the Amazon](https://www.youtube.com/watch?v=JTOWxi17k-w)
+- [Prezi Got pwned - A tale of responsible disclosure](http://engineering.prezi.com/blog/2014/03/24/prezi-got-pwned-a-tale-of-responsible-disclosure/)
+
+## Exploitation
```

Comment:
I'd like a bit more demo in this section (pictures etc) but I'll accept as is with the meta data changed above, and the request that you come back and make a pull request with some pictures once I make a path to do so.